lib/routes.js

```diff
@@ -358,7 +358,12 @@ module.exports = async function (app, opts) {
 
       return Promise.all(request.body.map(async (r, operationId) => {
         // Create individual reqs for multiple operations, otherwise reference the original req
-        const operationReq = operationsCount > 1 ? Object.create(request) : request
+        const operationReq = operationsCount > 1
+          ? {
+              ...request,
+              [kRequestContext]: Object.create(request[kRequestContext])
```

Comment:
Please note that this will do a shallow copy of `mercurius.requestContext`, which is all is needed to uniquely identify batched queries. If a deep clone is needed in the future we'll need to replace `Object.create` with a proper deep cloning function.

Reply:
I tried that combination, but [here they say](url) "once you throw in an empty object parameter to Object.assign(), the object spread operator is consistently faster", and I thought that would be beneficial for a complex object like the request.

I also tried many, many other combinations (trust me I did 🥲), and in the end this was the most readable and convenient I could find.